copier/copier.go

```diff
@@ -1400,6 +1428,31 @@ func copierHandlerGet(bulkWriter io.Writer, req request, pm *fileutils.PatternMa
 	return &response{Stat: statResponse.Stat, Get: getResponse{}}, cb, nil
 }
 
+func copyParentsDirs(name string, path string, alreadyCopied map[string]struct{}, copy func(string, string, os.FileInfo) error) (map[string]struct{}, error) {
+	parentName := filepath.Dir(name)
+	parentPath := filepath.Dir(path)
+	for parentName != "/" && parentName != "." {
+		if _, ok := alreadyCopied[parentPath]; ok {
+			parentName = filepath.Dir(parentName)
+			parentPath = filepath.Dir(parentPath)
+			continue
+		}
+
+		parentInfo, err := os.Lstat(parentPath)
+		if err != nil {
+			return alreadyCopied, fmt.Errorf("copier: get: lstat %q: %w", parentPath, err)
+		}
+		if err := copy(strings.TrimPrefix(parentName, "/"), parentPath, parentInfo); err != nil {
+			return alreadyCopied, err
+		}
+
+		alreadyCopied[parentPath] = struct{}{}
+		parentName = filepath.Dir(parentName)
+		parentPath = filepath.Dir(parentPath)
+	}
+	return alreadyCopied, nil
```

Comment:
This kind of implies that `alreadyCopied` wasn't already modified in the caller's scope, but this function was passed the map itself rather than a copy, so returning it here isn't really necessary.